### DISCUSSION TITLE
fipsintall to generate fips config file on the machine where application is installed.

### DISCUSSION THREAD

**Discussion by ChauhanPriyanka** [openssl/openssl#26021] [Q&A] [answered]
Hi team,

As per this article https://github.com/openssl/openssl/issues/17883#issuecomment-1469469383, fips.cnf file should be generated on the same machine/server where the application is installed.
So with 3.0.10 version of openssl , when I run following command, I see install_mac is generated, however it is not generated in openssl version 3.4. Is this expected or m I missing something.

command used
`openssl fipsinstall -module ./fips.dll -out fips.cnf -provider_name fips`

with openssl 3.0.10
```
[fips_sect]
activate = 1
install-version = 1
conditional-errors = 1
security-checks = 1
module-mac = 05:F0:AF:6B:D2:39:8E:D0:4C:1F:F4:5A:D4:CA:64:32:A8:FA:71:47:34:B6:DC:76:67:A8:42:48:75:F3:66:4B
install-mac = 41:9C:38:C2:8F:59:09:43:2C:AA:2F:58:36:2D:D9:04:F9:6C:56:8B:09:E0:18:3A:2E:D6:CC:69:05:04:E1:11
install-status = INSTALL_SELF_TEST_KATS_RUN
```

with openssl 3.4
```
[fips_sect]
activate = 1
install-version = 1
conditional-errors = 1
security-checks = 1
hmac-key-check = 0
kmac-key-check = 0
tls1-prf-ems-check = 0
no-short-mac = 0
drbg-no-trunc-md = 0
signature-digest-check = 0
hkdf-digest-check = 0
tls13-kdf-digest-check = 0
tls1-prf-digest-check = 0
sshkdf-digest-check = 0
sskdf-digest-check = 0
x963kdf-digest-check = 0
dsa-sign-disabled = 0
tdes-encrypt-disabled = 0
rsa-pkcs15-pad-disabled = 0
rsa-pss-saltlen-check = 0
rsa-sign-x931-pad-disabled = 0
hkdf-key-check = 0
kbkdf-key-check = 0
tls13-kdf-key-check = 0
tls1-prf-key-check = 0
sshkdf-key-check = 0
sskdf-key-check = 0
x963kdf-key-check = 0
x942kdf-key-check = 0
pbkdf2-lower-bound-check = 1
ecdh-cofactor-check = 0
module-mac = 43:70:B2:88:09:BB:4D:39:16:9E:5C:0D:FF:01:8F:7E:8D:1D:DD:E1:3A:D5:98:27:80:51:6B:30:BF:34:8C:F2
```

**Comment by t8m** [comment, marked as answer]
> So with 3.0.10 version of openssl , when I run following command, I see install_mac is generated, however it is not generated in openssl version 3.4. Is this expected or m I missing something.

Yes, 3.4 version complies with FIPS 140-3 and that does not allow skipping the self tests on subsequent start ups after the installation. Thus running openssl fipsinstall on the installation target machine is not required with 3.4 FIPS module.

**Reply by ChauhanPriyanka** [reply]
Thank you for the response. So now we can supply fips.cnf file along with the application, i.e. generate once on build machine and distribute across.

**Reply by t8m** [reply]
Yes. But please note that no other versions of the fips provider than 3.0.9 and 3.0.8 are yet validated.
